### PR TITLE
John Morrow

### PR DESCRIPTION
**Pull request by CodeMorrow**

### CODE REVIEW

File: test_4_filter_numbers.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 def filter_numbers(arr):
   # Remove the `return None` code below and add your own code.
-  return None
+  for i in arr[:]:
```
Comment (halfghaninne) [inline test_4_filter_numbers.py:3]: Can you tell me more about your use of `arr[:]`? What was your thought process in including it?

**Comment by CodeMorrow** [comment]
I was going to use a while loop to check the entire array like this:

`while (i<len(arr)` 

but as I researched through google I stumbled upon arr[:].  It is called Slicing. I could start the list at arr[5] and go to the end of the list by using arr[5:] or the opposite and go from the first item in the list to arr[4] by using arr[:5](this doesn't included arr[5]). By using arr[:] I select the entire list and use no slicing.  I thought it was a pretty neat way of attacking this problem.

**Comment by halfghaninne** [comment]
I really like your `for` loop _for_ sure (heh, heh). What I'm trying to understand is if slicing is necessary at all. Are `arr[:]` and `arr` equal?

**Comment by CodeMorrow** [comment]
I tried to run the code with just `arr` as well and it did not pass the test. I guess I used the slicing method for a slightly different purpose than intended for. I found it helpful to be able to select the entire list, no matter the length, this way.

**Pull request closed without merging**